### PR TITLE
LLDB debug configuration

### PR DESCRIPTION
**Pull request by garuma**
### Motivation

This PR adds some integration when working with LLDB.

Firstly, it creates an .lldbinit file which can be sourced to set up debug sessions properly. Specifically, it will load a Python script that will compute proper source mappings as Bazel builds all binaries from a username-specific cache location. Thus for LLDB output to make sense we need to map that path back to our main sources.

Secondly, it adds a sample VSCode debug launch.json to be able to debug LSP unit tests (which can be easily duplicated for other runners). Of note are the the two key additions to the configuration from the default templates aka `setupCommands` and `sourceFileMap`. The first one make sure that .lldbinit is properly sourced in the session (which will set a proper source map at the LLDB level). The second one will ensure that the VSCode *UI* correctly maps breakpoints/stacktrace information from LLDB to the workspace (why would it be not reading the LLDB source map setting and using that? no clue).

Note: depends on https://github.com/sorbet/sorbet/pull/3515

### Test plan

No tests

### CODE REVIEW

**Comment by garuma** [comment]
@jvilk-stripe based on what we discussed

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "LSP Test - Debug active test",
```
Comment (jvilk-stripe) [inline .vscode/launch.json:5]: Ohh. We should do this for CLI test too (can be in separate PR if you want).

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "${workspaceFolder}/bazel-bin/test/lsp_test_runner",
+            "args": ["--single_test=${fileDirname}/${fileBasenameNoExtension}", "--dt-no-colors"],
+            "stopAtEntry": false,
+            "cwd": "${workspaceFolder}",
+            "environment": [],
+            "externalConsole": false,
+            "MIMode": "lldb",
```
Comment (jvilk-stripe) [inline .vscode/launch.json:14]: 🤔  this isn't the third-party VS Code LLDB extension, looks like it's the official VS Code LLDB support discussed here:
https://code.visualstudio.com/docs/cpp/launch-json-reference

Does this require the following not-recommended extension (I hope not)?
https://github.com/sorbet/sorbet/blob/master/.vscode/extensions.json#L8

**Review by jvilk-stripe — APPROVED** [review]
My only concern is: Does this require Microsoft's C++ extension for VS Code? That extension blows up on Sorbet (and eats CPU for breakfast). If so, we should find a way to calm/disable its intellisense via settings.

File: .vscode/launch.json (hunk 1 above)
Comment (garuma) [reply .vscode/launch.json:5]: Absolutely, I'm now working on testing debug visualizer support for some data type but I can add some extra similar configs for other runners

File: .vscode/launch.json (hunk 2 above)
Comment (garuma) [reply .vscode/launch.json:14]: As this been revisited since that was put in place in 2019? I have been using the MS C++ support without issue as of now (no CPU issue at rest from what I can tell)
